eclipse-che/che-plugin-registry
# Install VS Code extensions by link

### What does this PR do?
Improve list of vsix files to be published in local openvsx registry. With these changes the extension could be described with this  schema:
```
{
  "id": "publisher.name" -- required,
  "download": "url_to_downlod_vsix" -- optional,
  "version": "extension_version" -- optional
}
```
If **download** attribute is not set, the script will try to fetch a vsix file from openvsx.org
If **version** is not set, the latest version of an extension will be downloaded.

Add GH PR check to validate the content of openvsx-sync.json file

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/21818

### How to test this PR?

Add a new extension into openvsx-sync.json. For example: 
```
    {
        "id":"GoogleCloudTools.cloudcode",
        "download":"https://GoogleCloudTools.gallery.vsassets.io/_apis/public/gallery/publisher/GoogleCloudTools/extension/cloudcode/1.20.3/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage",
        "version":"1.20.3"
    }
```
Build the image and check that GoogleCloudTools.cloudcode was published into the local openvsx registry.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
